lmnet/executor/build_tfds.py

```diff
@@ -0,0 +1,153 @@
+        raise NotImplementedError("A dataset builder for segmentation dataset is not implemented yet.")
+
+    if issubclass(dataset_class, ObjectDetectionBase):
+        return ObjectDetectionBuilder
+
+    return ClassificationBuilder
+
+
+def _copy_directory_recursively(src, dst):
+    tf.io.gfile.makedirs(dst)
+    for parent, directories, files in tf.io.gfile.walk(src):
+        for directory in directories:
+            src_dir = os.path.join(parent, directory)
+            dst_dir = os.path.relpath(src_dir, src)
+            dst_dir = os.path.join(dst, dst_dir)
+
+            if tf.io.gfile.exists(dst_dir):
+                tf.io.gfile.rmtree(dst_dir)
+
+            tf.io.gfile.mkdir(dst_dir)
+
+        for file in files:
+            src_file = os.path.join(parent, file)
+            dst_file = os.path.relpath(src_file, src)
+            dst_file = os.path.join(dst, dst_file)
+            tf.io.gfile.copy(src_file, dst_file)
+
+
+def run(config_file, overwrite):
+    """Build custom TFDS datasets from config file"""
+    dataset_class, dataset_kwargs, dataset_name, data_dir = _get_tfds_settings(config_file)
+
+    if not overwrite and tf.io.gfile.exists(os.path.join(data_dir, dataset_name)):
+        raise ValueError("Output path already exists: {}\n"
+                         "Please use --overwrite if you want to overwrite."
+                         .format(os.path.join(data_dir, dataset_name)))
+
+    builder_class = _get_tfds_builder_class(dataset_class)
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        builder = builder_class(dataset_name=dataset_name,
+                                dataset_class=dataset_class,
+                                dataset_kwargs=dataset_kwargs,
+                                data_dir=tmpdir)
+
+        builder.download_and_prepare()
+        print("Dataset was built successfully.")
+
+        print("Copying to destination...")
+        _copy_directory_recursively(src=tmpdir, dst=data_dir)
+
+        print("Done!!")
+
+
+@click.command(context_settings=dict(help_option_names=["-h", "--help"]))
+@click.option(
+    "-c",
+    "--config_file",
+    help="A path to config file",
+    required=True,
+)
+@click.option(
+    "-o",
+    "--overwrite",
+    help="Overwrite if the output directory already exists.",
+    is_flag=True,
+    default=False,
+)
+def main(config_file, overwrite):
+    """
+    A script to build custom TFDS datasets
+
+    \b
+    This script can build custom TFDS datasets from existing dataset classes.
+    You can use training config files to specify which dataset class to be used as data source.
+    The following settings are required in the config file.
+
+    \b
+    ```
+    DATASET_CLASS = <dataset class>
+    DATASET.TFDS_KWARGS = {
+        "name": "<a dataset name to be generated>",
+        "data_dir": "<a directory path to output generated dataset>",
+        "image_size": <image size array like [128, 128]>,
```

Comment:
We should describe when we apply this `image_size` into image, loading from TFRecord or making TFRecord. 🤔  Because it's information for making a decision of exectuting this script (creating TFRecord) when developer training models.

Reply:
I've added comments.